gamemode/modules/base/sv_gamemode_functions.lua

```diff
@@ -304,36 +304,40 @@ local vrad = GM.Config.voiceradius
 local dynv = GM.Config.dynamicvoice
 local deadv = GM.Config.deadvoice
 local voiceDistance = GM.Config.voiceDistance * GM.Config.voiceDistance
+local DrpCanHear = {}
 -- proxy function to take load from PlayerCanHearPlayersVoice, which is called a quadratic amount of times per tick,
 -- causing a lagfest when there are many players
 local function calcPlyCanHearPlayerVoice(listener)
     if not IsValid(listener) then return end
-    listener.DrpCanHear = listener.DrpCanHear or {}
+    DrpCanHear[listener] = DrpCanHear[listener] or {}
+    if listener:IsBot() then return end
     local shootPos = listener:GetShootPos()
-    for _, talker in ipairs(player.GetAll()) do
+    for _, talker in ipairs(player.GetHumans()) do
         local talkerShootPos = talker:GetShootPos()
-        listener.DrpCanHear[talker] = not vrad or -- Voiceradius is off, everyone can hear everyone
+        DrpCanHear[listener][talker] = not vrad or -- Voiceradius is off, everyone can hear everyone
             (shootPos:DistToSqr(talkerShootPos) < voiceDistance and -- voiceradius is on and the two are within hearing distance
                 (not dynv or IsInRoom(shootPos, talkerShootPos, talker))) -- Dynamic voice is on and players are in the same room
     end
 end
 hook.Add("PlayerInitialSpawn", "DarkRPCanHearVoice", function(ply)
+    calcPlyCanHearPlayerVoice(ply)
+    if ply:IsBot() then return end
     timer.Create(ply:UserID() .. "DarkRPCanHearPlayersVoice", 0.5, 0, fn.Curry(calcPlyCanHearPlayerVoice, 2)(ply))
 end)
+
 hook.Add("PlayerDisconnected", "DarkRPCanHearVoice", function(ply)
-    if not ply.DrpCanHear then return end
-    for _, v in ipairs(player.GetAll()) do
-        if not v.DrpCanHear then continue end
```

Comment:
It was this check I was talking about, though, not the one in `calcPlyCanHearPlayerVoice`. Sorry if I didn't make that clear earlier.

Reply:
In order to fuck this one up, PlayerInitialSpawn HAS to error and then only the part with removing OTHER players from your table can cause errors.

Reply:
The rest of the thing, physically can't error. It just can't now.